cpp/Platform.Converters/CachingConverterDecorator.h

```diff
@@ -1,15 +1,38 @@
-﻿namespace Platform::Converters
+﻿#pragma once
+#include <map>
+#include <memory>
+
+#include "IConverter[TSource, TTarget].h"
+
+namespace Platform::Converters
 {
     template <typename ...> class CachingConverterDecorator;
     template <typename TSource, typename TTarget> class CachingConverterDecorator<TSource, TTarget> : public IConverter<TSource, TTarget>
     {
-        private: readonly IConverter<TSource, TTarget> *_baseConverter;
-        private: readonly IDictionary<TSource, TTarget> *_cache;
+        private:
+          std::unique_ptr<IConverter<TSource, TTarget>> _baseConverter;
+          std::unique_ptr<std::map<TSource, TTarget>> _cache;
 
-        public: CachingConverterDecorator(IConverter<TSource, TTarget> &baseConverter, IDictionary<TSource, TTarget> &cache) { (_baseConverter, _cache) = (baseConverter, cache); }
+        private:
+          CachingConverterDecorator(IConverter<TSource, TTarget>& baseConverter, std::map<TSource, TTarget>& cache)
+            : _baseConverter(baseConverter)
+            , _cache(cache)
+          {}
 
-        public: CachingConverterDecorator(IConverter<TSource, TTarget> &baseConverter) : this(baseConverter, Dictionary<TSource, TTarget>()) { }
+          CachingConverterDecorator(IConverter<TSource, TTarget> &baseConverter)
+            : _baseConverter(baseConverter)
+          {
+            _cache = std::make_unique<std::map<TSource, TTarget>>();
+          }
 
-        public: TTarget Convert(TSource source) { return _cache.GetOrAdd(source, _baseConverter.Convert); }
+          TTarget Convert(TSource source)
+          {
+            auto iter = _cache->find(source);
```

Comment:
Please use consistent indentation. Always 4 spaces per level.
Use `cursor` name for the iterator's variable here, please.

Reply:
Ok